### ISSUE TITLE
StickerService.js Visual Studio Code Additions.

### ISSUE THREAD

**Issue by JohnEdwa** [doki-theme/doki-theme-vscode#59]
First of all, I'm sorry for doing this as an issue, but by making the project construct itself from ten different places means I have no idea where or even how I should make this as a pull request. At least it is a VScode thing, so it's the correct repo.

---

Anyway, I've been tuning the StrickerService.js wallpaper installation script to make more stuff transparent/wallpapered, and here are my changes to it. I kept them separated so it would be easier to handle, but any class with "background-image: url('${wallpaperURL}') !important;" are identical to the one already in the CSS.

Makes the welcome screen transparent
![welcomescreen](https://user-images.githubusercontent.com/5417183/114317040-b9373480-9af5-11eb-8021-79fd1dfa918a.png)

```
.editor-container {
	background-image: url('${wallpaperURL}') !important;
	/* blah */
}

.editor-instance,
.welcomePageContainer,
.welcomePage,
.welcomePageFocusElement {
	background-color: transparent !important;
}
```

---

Fix "wallpaper" being drawn over "background", mostly only important if they differ from each other.  

![split-view-view](https://user-images.githubusercontent.com/5417183/114317061-ceac5e80-9af5-11eb-8727-ed474e58f1f6.png)

```
/* Remove from the CSS that gives it a background */
	.split-view-view {
		background-color: transparent !important; /* Just in case. */
	}
```

---

"OUTPUT" panel is not transparent. 

```
.monaco-workbench .part.panel > .content .monaco-editor .monaco-editor-background {
	background-image: url('${wallpaperURL}') !important;
	/* blah */
}
```

Left margin of the TERMINAL window

```
div.terminal-outer-container  {
	background-image: url('${wallpaperURL}') !important;
	/* blah */
}
```

---

DEBUG CONSOLE and warning/error popups
![image](https://user-images.githubusercontent.com/5417183/114317500-ea186900-9af7-11eb-81f5-fbf05b848f6c.png)

```
div.monaco-scrollable-element > div.monaco-list-rows{
	background-image: url('${wallpaperURL}') !important;
	/* blah */
}
```

---

So in full, line 33 onwards looks like this:

```  
[id="workbench.parts.editor"] .split-view-view .editor-container .editor-instance>.monaco-editor  .overflow-guard>.monaco-scrollable-element::before,
.overflow-guard, 
.tab, 
.tabs-container,
.monaco-pane-view,
.composite.title, 
.content, 
.monaco-select-box, 
.pane-header, 
.xterm-cursor-layer,
.minimap-decorations-layer,
.decorationsOverviewRuler,
.monaco-workbench .part.editor>.content .editor-group-container>.title .tabs-breadcrumbs .breadcrumbs-control,
.ref-tree, /* find usages */
.head, /* find usages */
.monaco-workbench .part.editor>.content .editor-group-container>.title .editor-actions,  
/* .split-view-view, */
.editor-container,
.monaco-workbench .part.panel > .content .monaco-editor .monaco-editor-background,
div.monaco-scrollable-element > div.monaco-list-rows,
div.terminal-outer-container  
{
	background-image: url('${wallpaperURL}') !important;
	background-position: ${backgroundAnchoring} !important;
	background-attachment: fixed !important;
	background-repeat: no-repeat !important;
	background-size: cover !important;
}

/* transparencies */
.editor-instance,
.welcomePageContainer,
.welcomePage,
.welcomePageFocusElement, 
{
	background-color: transparent !important;
}
```

**Comment by Unthrottled** [comment]
Aight, so I am a bit confused here. Let me see if I understand what you are trying to accomplish.

- There are some elements that don't conform `glass pane` wallpaper feature, and you want to fix that.
- You've been modifying the javascript transpiled from the `StickerService.ts`
- You wan't these changes to be part of the plugin's source code

Sound about right?

**Comment by JohnEdwa** [comment]
Yes, Yes, and Yes, in case you think they should be.

Also I seem to be blind because I somehow wasn't able to find that .ts file.

**Comment by Unthrottled** [comment]
> DEBUG CONSOLE and warning/error popups
>![image](https://user-images.githubusercontent.com/5417183/114317500-ea186900-9af7-11eb-81f5-fbf05b848f6c.png)
>
>```
>div.monaco-scrollable-element > div.monaco-list-rows{
>	background-image: url('${wallpaperURL}') !important;
>	/* blah */
>}
>```

:point_up: Not a big fan of this. I've accidentally did that, and didn't quite care for it. It breaks the illusion that the image is behind a transparent pane of glass. Let me try something out on the same lines and I'll see if I like it. Something like this https://css-tricks.com/frosting-glass-css-filters/

Plus it also does this
![Peek 2021-04-11 17-59](https://user-images.githubusercontent.com/15972415/114324485-d832c800-9aef-11eb-829c-94b04fceb865.gif)

**Comment by Unthrottled** [comment]
Oh yeah, [this is the typscript code](https://github.com/doki-theme/doki-theme-vscode/blob/master/src/StickerService.ts)

**Comment by JohnEdwa** [comment]
Ah, didn't notice that issue.
Honestly the popups were an accident as I was trying to do the debug box, and noticed afterwards that it also changed those. I'll see if I can find some other way for those, and do a proper pull request.

**Comment by JohnEdwa** [comment]
>It breaks the illusion that the image is behind a transparent pane of glass.

There is always `backdrop-filter: blur(1px)`:

![image](https://user-images.githubusercontent.com/5417183/114325385-670b0900-9b1f-11eb-9b1a-acf4396bcf94.png)

Annoyingly it also requires the background colour in RGBA to work properly as it needs to be set partly opaque: `background-color: rgba(37, 37, 38, 0.2) !important;`

**Comment by Unthrottled** [comment]
:heart_eyes: I really like that

> Annoyingly it also requires the background colour in RGBA to work properly as it needs to be set partly opaque: background-color: rgba(37, 37, 38, 0.2) !important;

So that would need to be as part of the wallpaper css? (I assume  rgba(37, 37, 38, 0.2) is the converted value of the original background hex)

**Comment by Unthrottled** [comment]
Closed by #66
